src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+				Id:      d.GroupId,
+				Members: members,
+			})
+		}
+		bytes, err := json.Marshal(&kafkaMetadata{
+			Brokers:        brokers,
+			Topics:         topics,
+			ConsumerGroups: cgs,
+		})
+		if err != nil {
+			return fmt.Errorf("couldn't marshal Kafka metadata: %w", err)
+		}
+		return writeFileToZip(ps, "kafka.json", bytes)
+	}
+}
+
+// TODO: filter by time or size (GiB)
+// TODO: What if running inside a container/ k8s?
+func saveLogs(ps *stepParams, since, until string) step {
+	return func() error {
+		args := []string{"--no-pager", "-u", "redpanda"}
+		if since != "" {
+			args = append(args, "--since", since)
+		}
+		if until != "" {
+			args = append(args, "--until", until)
+		}
+		return writeCommandOutputToZip(
+			ps,
+			"redpanda.log",
+			"journalctl",
+			args...,
+		)
+	}
+}
+
+func saveConfig(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		bs, err := yaml.Marshal(conf)
+		if err != nil {
+			return fmt.Errorf("couldn't encode the redpanda config as YAML: %w", err)
+		}
+		return writeFileToZip(ps, "redpanda.yaml", bs)
+	}
+}
+
+func saveResourceUsageData(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		res, err := system.GatherMetrics(ps.fs, ps.timeout, *conf)
+		if system.IsErrRedpandaDown(err) {
+			return fmt.Errorf("omitting resource usage metrics: %w", err)
+		}
+		if err != nil {
+			return fmt.Errorf("error gathering resource usage metrics: %w", err)
+		}
+		bs, err := json.Marshal(res)
+		if err != nil {
+			return fmt.Errorf("couldn't encode resource usage metrics: %w", err)
+		}
+		return writeFileToZip(ps, "resource-usage.json", bs)
+	}
+}
+
+func saveNTPDrift(ps *stepParams) step {
+	return func() error {
+		u, err := user.Current()
+		if err != nil {
+			return err
+		}
+		if u.Uid != "0" {
+			return errors.New(
+				"omitting NTP drift (clockdiff) data since it wasn't run as" +
+					" root (it requires the CAP_NET_RAW capability)",
+			)
+		}
+		return writeCommandOutputToZip(
+			ps,
+			"clockdiff.txt",
+			"clockdiff",
+			"'pool.ntp.org'",
+		)
+	}
+}
+
+func saveDNSData(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(ps, "dig.txt", "dig")
+	}
+}
+
+func saveSocketData(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(ps, "ss.txt", "ss")
+	}
+}
+
+func saveTopOutput(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"top.txt",
+			"top", "-n", "1",
+		)
+	}
+}
+
+func saveVmstat(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"vmstat.txt",
+			"vmstat",
+		)
+	}
+}
+
+func saveIfconfig(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"ifconfig.txt",
+			"ifconfig",
+		)
+	}
+}
+
+func saveDataDirStructure(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"data-dir-tree.txt",
+			"tree",
```

Comment:
`find` might be more ubiquitous. I don't have tree on my system.

Reply:
Good tip, thanks!

Reply:
Wouldn’t this be more robust if we didn’t use shell for simple things like recursive directory traversal. Effectively a tree traversal

Reply:
Removed and reimplemented by doing a dir walk in Go, that also follows symlinks that point to dirs.